### ISSUE TITLE
Is BatchNorm momentum=None simple average mode supported?

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#31498]
SyncBatchNorm (and probably regular BatchNorm as well) docs say that momentum=None is equivalent to a cumulative average update of running stats.

However, [its impl](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Normalization.cpp#L618) doesn't seem to handle this special case.

This mode is useful for [SWA](https://github.com/pytorch/pytorch/pull/30559) for adapting BN running stats to a averaged model weights.

**Comment by ngimel** [comment]
cc @jjsjann123, do you know if SyncBatchNorm properly handles `momentum=None` case?

**Comment by vadimkantorov** [comment]
Same question about regular BatchNorm... if it’s properly supported, SWA code can be greatly simplified

**Comment by jjsjann123** [comment]
`momentum=None` is handled in the python side: 
https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/batchnorm.py#L90-L102
https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/batchnorm.py#L441-L451

**Comment by vadimkantorov** [comment]
Thanks! I hope it then can be used in SWA!

**Comment by tstandley** [comment]
@vadimkantorov Can you explain what a momentum parameter of none does differently? "simple average" and "cumulative moving average" both could mean different things, and the documentation doesn't disambiguate.

**Comment by vadimkantorov** [comment]
Default mode uses momentum and thus is a form of exponential moving average. None is a simple average mode. But I'm not a core dev :)
